src/search/search.py

```diff
@@ -18,69 +21,181 @@
+            load_placeholder_thumbnail(index, search_result.id)
+            index += 1
+            continue
 
-        Returns:
-            List: search results
-        """
-        if f'{HANA3D_NAME}_search_results' not in self.context.window_manager:
-            return []
-        return self.context.window_manager[f'{HANA3D_NAME}_search_results']
+        thumbnail_path = os.path.join(directory, search_result.thumbnail_small)
 
-    @property
-    def props(self):
-        """Get search props.
+        image_name = utils.previmg_name(index)
 
-        Returns:
-            Any | None: search props if available
-        """
-        if self.asset_type == 'model' and hasattr(self.context.window_manager, HANA3D_MODELS):  # noqa : WPS421
-            return getattr(self.context.window_manager, HANA3D_MODELS)
-        elif self.asset_type == 'scene' and hasattr(self.context.window_manager, HANA3D_SCENES):  # noqa : WPS421
-            return getattr(self.context.window_manager, HANA3D_SCENES)
-        elif self.asset_type == 'material' and hasattr(self.context.window_manager, HANA3D_MATERIALS):  # noqa : WPS421
-            return getattr(self.context.window_manager, HANA3D_MATERIALS)
+        if os.path.exists(thumbnail_path):  # sometimes we are unlucky...
+            img = bpy.data.images.get(image_name)
+            if img is None:
+                img = bpy.data.images.load(thumbnail_path)
+                img.name = image_name
+            elif img.filepath != thumbnail_path:
+                # had to add this check for autopacking files...
+                if img.packed_file is not None:
+                    img.unpack(method='USE_ORIGINAL')
+                img.filepath = thumbnail_path
+                img.reload()
+            img.colorspace_settings.name = 'Linear'
+        index += 1
+
+
+def load_placeholder_thumbnail(index: int, asset_id: str):
+    """Load placeholder thumbnail for assets without one.
+
+    Parameters:
+        index: index number of the asset in search results
+        asset_id: asset id
+    """
+    placeholder_path = paths.get_addon_thumbnail_path('thumbnail_notready.png')
+
+    img = bpy.data.images.load(placeholder_path)
+    img.name = utils.previmg_name(index)
+
+    hidden_img = bpy.data.images.load(placeholder_path)
+    hidden_img.name = f'.{asset_id}'
+
+    fullsize_img = bpy.data.images.load(placeholder_path)
+    fullsize_img.name = utils.previmg_name(index, fullsize=True)
+
+
+def get_search_results(asset_type: AssetType = None) -> List[SearchResult]:
+    """Get search results.
+
+    Parameters:
+        asset_type: type of the assets searched
+
+    Returns:
+        List: search results
+    """
+    if asset_type is None:
+        asset_type = _get_asset_type_from_ui()
+    if f'{HANA3D_NAME}_{asset_type}_search' not in bpy.context.window_manager:
+        return []
+    return [
+        SearchResult(**json.loads(json.dumps(props.to_dict())))
```

Comment:
Acho que a ideia é na verdade transformar de `SearchResult` para `dict`, então teria que usar algo tipo `vars()`.
https://stackoverflow.com/questions/61517/python-dictionary-from-an-objects-fields

Reply:
Mas aí se o retorno do `get` for um dict, a gente volta a não ter tipagem bem definida pros atributos dele
O que estou fazendo agora é:
- get: transforma de bpy.props para dict e retorna uma lista de SearchResult
- set: transforma de SearchResult para dict (usando um equivalente do vars)

Vou ver se consigo resolver os problemas das travadas/thumbnails ainda com o SearchResult, mas em último caso volto pra dict mesmo

Reply:
Bem, no caso então acho que não tem como fugir de ter que trocar todos os casos que citei no comentário acima. Que deve ser bastante coisa.

Reply:
Acho que os `search_results` não precisam ser armazenados no `window_manager`, poderia colocar tipo em um singleton, que facilitaria todo esse fluxo.

Reply:
Realmente, parece uma solução melhor